client/my-sites/people/team-list/index.jsx

```diff
@@ -1,35 +1,80 @@
 /**
  * External dependencies
  */
-import { localize } from 'i18n-calypso';
+import { useTranslate } from 'i18n-calypso';
 import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { errorNotice, removeNotice } from 'calypso/state/notices/actions';
 
 /**
  * Internal dependencies
  */
-import SiteUsersFetcher from 'calypso/components/site-users-fetcher';
 import Team from './team';
+import useUsers from 'calypso/data/users/use-users';
 
-class TeamList extends React.Component {
-	static displayName = 'TeamList';
+const useErrorNotice = ( error, refetch ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
 
-	render() {
-		const fetchOptions = {
-			siteId: this.props.site && this.props.site.ID,
-			order: 'ASC',
-			order_by: 'display_name',
-			search: this.props.search ? '*' + this.props.search + '*' : null,
-			search_columns: [ 'display_name', 'user_login' ],
-		};
+	React.useEffect( () => {
+		if ( ! error ) {
+			return;
+		}
 
-		Object.freeze( fetchOptions );
-
-		return (
-			<SiteUsersFetcher fetchOptions={ fetchOptions }>
-				<Team { ...this.props } />
-			</SiteUsersFetcher>
+		dispatch(
+			errorNotice( translate( 'There was an error retrieving users' ), {
+				id: 'site-users-notice',
+				button: 'Try again.',
+				onClick: () => {
+					dispatch( removeNotice( 'site-users-notice' ) );
+					refetch();
+				},
+			} )
 		);
-	}
+	}, [ dispatch, translate, error, refetch ] );
+};
+
+function TeamList( props ) {
+	const { site, search } = props;
+	const fetchOptions = search
+		? {
+				search: `*${ search }*`,
+				search_columns: [ 'display_name', 'user_login' ],
+		  }
+		: {};
+	const listKey = [ 'team', site.ID, search ].join( '-' );
+
+	const {
+		data,
+		isLoading,
+		isFetchingNextPage,
+		hasNextPage,
+		fetchNextPage,
+		error,
+		refetch,
+	} = useUsers( site.ID, fetchOptions );
+
+	useErrorNotice( error, refetch );
+
+	return (
+		<Team
+			listKey={ listKey }
+			fetchingUsers={ isLoading }
+			fetchingNextPage={ isFetchingNextPage }
+			totalUsers={ data?.total }
+			users={ data?.users ?? [] }
+			excludedUsers={ [] }
+			fetchNextPage={ fetchNextPage }
+			hasNextPage={ hasNextPage }
+			{ ...props }
```

Comment:
What if we were explicit about what props we are passing through here? It should be just `search` and `site`.

Reply:
Done via https://github.com/Automattic/wp-calypso/pull/50712/commits/9aedcf3a65adafde4b8b1dead3bf529765b1f1f9